packages/fetcher/src/fetcher.ts

```diff
@@ -16,226 +18,155 @@ import type {
 
     if (!result.ok) throw new FetcherError(result as FetcherResponse<undefined>)
 
     return result
   }
 
-  /**
-   * Makes a `GET` to a target `endpoint` with the provided `query` params.
-   */
-  public get<T>(
-    endpoint: string,
-    opts?: FetcherRequestOpts,
-  ): Promise<FetcherResponse<T>> {
-    return this.request<T>({
+  const get = async <T>(endpoint: string, opts?: FetcherRequestOpts) => {
+    return request<T>({
       endpoint,
       method: HTTP_METHODS.GET,
       ...(opts || {}),
     })
   }
 
-  /**
-   * Makes a `POST` to a target `endpoint` with the provided `body`.
-   */
-  public post<T>(
-    endpoint: string,
-    opts?: FetcherRequestOpts,
-  ): Promise<FetcherResponse<T>> {
-    return this.request<T>({
+  const post = async <T>(endpoint: string, opts?: FetcherRequestOpts) => {
+    return request<T>({
       endpoint,
       method: HTTP_METHODS.POST,
       ...(opts || {}),
     })
   }
 
-  /**
-   * Makes a `PATCH` to a target `endpoint` with the provided `body`.
-   */
-  public patch<T>(
-    endpoint: string,
-    opts?: FetcherRequestOpts,
-  ): Promise<FetcherResponse<T>> {
-    return this.request<T>({
+  const patch = async <T>(endpoint: string, opts?: FetcherRequestOpts) => {
+    return request<T>({
       endpoint,
       method: HTTP_METHODS.PATCH,
       ...(opts || {}),
     })
   }
 
-  /**
-   * Makes a `PUT` to a target `endpoint` with the provided `body`.
-   */
-  public put<T>(
-    endpoint: string,
-    opts?: FetcherRequestOpts,
-  ): Promise<FetcherResponse<T>> {
-    return this.request<T>({
+  const put = async <T>(endpoint: string, opts?: FetcherRequestOpts) => {
+    return request<T>({
       endpoint,
       method: HTTP_METHODS.PUT,
       ...(opts || {}),
     })
   }
 
-  /**
-   * Makes a `DELETE` to a target `endpoint` with the provided `body`.
-   */
-  public delete<T>(
-    endpoint: string,
-    opts?: FetcherRequestOpts,
-  ): Promise<FetcherResponse<T>> {
-    return this.request<T>({
+  const _delete = async <T>(endpoint: string, opts?: FetcherRequestOpts) => {
+    return request<T>({
       endpoint,
       method: HTTP_METHODS.DELETE,
       ...(opts || {}),
     })
   }
 
-  async #executeRequest(id: string, request: FetcherRequest) {
-    const config = await this.#interceptorManager.runRequestInterceptors(
-      request,
-    )
-    const timeoutId = setTimeout(() => this.abort(id), this.#config.timeout)
+  const executeRequest = async (id: string, request: FetcherRequest) => {
+    const cfg = await interceptorManager.runRequestInterceptors(request)
+    const timeoutId = setTimeout(() => abort(id), config.value.timeout)
 
-    const response = await fetch(config.url, config)
+    const response = await fetch(cfg.url, cfg)
 
-    this.#abortManager.clear(id)
+    abortManager.clear(id)
     clearTimeout(timeoutId)
     return response
   }
+
+  const fetcher = toRaw({
+    config,
+    baseUrl: computed(() => config.value.baseUrl),
```

Comment:
if it's the only place to use hook from reactivity lib, better to write just a small getter func and remove lib from dependencies